### ISSUE TITLE
jax.numpy.take() won't return the right value for out-of-bound indices in version 0.4.11

### ISSUE THREAD

**Issue by RongdeQiu** [google/jax#16899]
### Description

0.4.11						print(jax.__version__)
[ 0.  1.  2.  3.  4. nan nan]	<----print(jnp.take(jnp.arange(5).astype(float), jnp.arange(7)))
[0. 1. 2. 3. 4. 4. 4.]		<----print(jnp.take(jnp.arange(5).astype(float), jnp.arange(7), mode='clip'))
[0. 1. 2. 3. 4. 4. 4.]		<----print(np.take(np.arange(5).astype(float), np.arange(7), mode='clip'))
Jax 0.4.11, jaxlib 0.4.10, without jax-metal

0.4.11						print(jax.__version__)
[0. 1. 2. 3. 4. 0. 0.]		<----print(jnp.take(jnp.arange(5).astype(float), jnp.arange(7)))
[0. 1. 2. 3. 4. 0. 0.]		<----print(jnp.take(jnp.arange(5).astype(float), jnp.arange(7), mode='clip'))
[0. 1. 2. 3. 4. 4. 4.]		<----print(np.take(np.arange(5).astype(float), np.arange(7), mode='clip'))
Jax 0.4.11, jaxlib 0.4.10, jax-metal 0.0.3

Looks like Jax-metal caused this

### What jax/jaxlib version are you using?

jax 0.4.11, jaxlib 0.4.10, Jax-metal 0.0.3

### Which accelerator(s) are you using?

GPU

### Additional system info

python 3.10.12

**Comment by jakevdp** [comment]
Thanks for the report - I'm not sure if this was closed on purpose, but note that the jax-metal plugin is experimental and still has a number of issues that make its outputs incompatible with JAX's behavior on other backends. You can find a list of known issues at the [Apple-GPU-Metal-plugin](https://github.com/google/jax/labels/Apple%20GPU%20%28Metal%29%20plugin) tag.
